src/Core/src/Platform/iOS/ViewExtensions.cs

```diff
@@ -650,22 +675,36 @@ internal static IDisposable OnUnloaded(this UIView uiView, Action action)
 				return new ActionDisposable(() => { });
 			}
 
-			Dictionary<NSString, NSObject> observers = new Dictionary<NSString, NSObject>();
+			var observers = new List<IDisposable>(2);
 			ActionDisposable? disposable = null;
 			disposable = new ActionDisposable(() =>
 			{
 				disposable = null;
 				foreach (var observer in observers)
 				{
-					uiView.Layer.RemoveObserver(observer.Value, observer.Key);
-					observers.Remove(observer.Key);
+					observer.Dispose();
 				}
+				observers.Clear();
 			});
 
 			// Ideally we could wire into UIView.MovedToWindow but there's no way to do that without just inheriting from every single
-			// UIView. So we just make our best attempt by observering some properties that are going to fire once UIView is attached to a window.	
-			observers.Add(new NSString("bounds"), (NSObject)uiView.Layer.AddObserver("bounds", Foundation.NSKeyValueObservingOptions.OldNew, (_) => UnLoadedCheck()));
-			observers.Add(new NSString("frame"), (NSObject)uiView.Layer.AddObserver("frame", Foundation.NSKeyValueObservingOptions.OldNew, (_) => UnLoadedCheck()));
```

Comment:
With the MovedToWindow you originally had the unloaded also listening to the bounds/frame, and now i see you removed it. Did it not work? Is there a case where we are now going to not get an unloaded?

Reply:
AFAICT it did not work. There's an issue about it here as well https://github.com/dotnet/maui/pull/16868 . 

I ran all of our UITests around loaded/unloaded and it was never triggered by the observers. It was only triggered by the logic that fires unloaded when an element is removed from the logical tree. 

> Is there a case where we are now going to not get an unloaded?

There might be some still, but I'd say there are less after this PR.